### DISCUSSION TITLE
BLT Not Auto Homing

### DISCUSSION THREAD

**Discussion by Sipheren** [Jyers/Marlin#439] [Q&A] [answered]
Hi,

So wanted to try this firmware out, I downloaded E3V2-UBL-BLTouch-15x15-HS-v4.2.7.bin and flashed it but now when I try to auto home (g28) it just errors out...

I don't really get what the deal is, was fine with my own firmware, is there something special that isn't noted and needs to be done for it to work?

https://sipheren.com/stuff/blt.mp4

**Comment by Sipheren** [comment, marked as answer]
Figured it out, guessing I have the BLTouch connected a bit diff (followed Teaching Techs Guide). So I had to make some changes:

I need this:

#define Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN


The pre-built ones use this instead:

// Force the use of the probe for Z-axis homing
#define USE_PROBE_FOR_Z_HOMING

Cheers

**Comment by AllPointsSouth** [comment]
I'm having the same issue, please can you share your build

**Reply by Sipheren** [reply]
Hey,

So I used this guide when I upgraded to a 4.2.7 board and used his wiring method as I find it a simpler way.

https://youtu.be/neS7lB7fCww

You can download the firmware I made here, its using UBL, 10x10, BLTouch for 4.2.7 based on the source files I downloaded from Jyers.

https://sipheren.com/stuff/E3V2-UBL-BLTouch-10x10-HS-v4.2.7.bin
